src/tests/system/tests/test_passkey.py

```diff
@@ -420,3 +420,81 @@ def test_passkey__su_same_key_for_multi_user(
             ioctl=f"{moduledatadir}/umockdev.ioctl",
             script=f"{testdatadir}/umockdev.script.{suffix}.{user}",
         )
+
+
+@pytest.mark.importance("high")
+@pytest.mark.ticket(jira="SSSD-7011", gh=7066)
+@pytest.mark.topology(KnownTopologyGroup.AnyAD)
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.builtwith(client="passkey", provider="passkey")
+def test_passkey__ssh_key_as_passkey_mapping_token(
+    client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str
+):
+    """
+    :title: Check ssh key as a passkey mapping token with AD, Samba and LDAP server.
+    :setup:
+        1. Add a users in AD, Samba and LDAP server and add ssh key as a passkey mapping.
+        2. Setup SSSD client with FIDO, start SSSD service.
+    :steps:
+        1. Check su non-passkey authentication of the user.
+        2. Required error message in pam log.
+    :expectedresults:
+        1. su failed to authenticate the user with correct password.
+        2. Get the expected message in pam log.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+
+    client.sssd.domain["local_auth_policy"] = "only"
+
+    with open(f"{testdatadir}/ssh-key") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    client.sssd.start()
+
+    # We are running simple su not to check authentication with passkey but just to get
+    # expected log message.
+    assert not client.auth.su.password("user1", "Secret123"), "Authenticate the user with correct password"
+
+    pam_log = client.fs.read(client.sssd.logs.pam)
+    assert "Mapping data found is not passkey related" in pam_log, "String was not found in the logs"
+
+
+@pytest.mark.importance("high")
+@pytest.mark.ticket(jira="SSSD-7011", gh=7066)
+@pytest.mark.topology(KnownTopologyGroup.AnyAD)
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.builtwith(client="passkey", provider="passkey")
+def test_passkey__ssh_key_with_passkey_mapping_token(
+    client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str
+):
+    """
+    :title: Check ssh key with a passkey mapping token with AD, Samba and LDAP server.
+    :setup:
+        1. Add a users in AD, Samba and LDAP server and add ssh key and a passkey mapping.
+        2. Setup SSSD client with FIDO, start SSSD service.
+    :steps:
+        1. Check su non-passkey authentication of the user.
+        2. Required error message in pam log.
+    :expectedresults:
+        1. su failed to authenticate the user with correct password.
```

Comment:
Do we expect the su to fail if the user has both an invalid (ssh) and valid (passkey) key mapped?  I would have thought SSSD would log a warning about an invalid mapping and use the valid one.

Reply:
yes, if we add `local-auth_policy=only` for all provider as discussed, things changed.
We can't log in with `password` nor with `passkey`.

Reply:
@justin-stephenson is checking, let him check, because after `local_auth_policy = only` auth is failing in both the cases.

Reply:
Hi @sumit-bose setting `local_auth_policy = only` breaks password authentication in a direct AD integration configuration. In SSS_PAM_AUTHENTICATE step of pam responder:

```
(2024-04-05 14:20:03): [pam] [pam_dom_forwarder] (0x3f7c0): [CID#1] Local auth only set, skipping online auth
(2024-04-05 14:20:03): [pam] [pam_reply] (0x4000): [CID#1] pam_reply initially called with result [4]: System error. this result might be changed during processing
(2024-04-05 14:20:03): [pam] [pam_reply] (0x0400): [CID#1] Local auth policy allowed: smartcard [True], passkey [True]
(2024-04-05 14:20:03): [pam] [pam_reply] (0x0200): [CID#1] blen: 24
(2024-04-05 14:20:03): [pam] [pam_reply] (0x0200): [CID#1] Returning [4]: System error to the client
(2024-04-05 14:20:04): [pam] [client_recv] (0x0200): [CID#1] Client disconnected!
```

IIRC this was expected based on how we designed `local_auth_policy`. Do you agree? If yes, I guess `local_auth_policy = enable:passkey` is the better approach when using multiple auth types such as what Madhuri is trying here ?

Reply:
Hi,

I would say the behavior is correct because password authentication with AD is initially not a local authentication method. Only if it is allowed to store the password hash locally and a successful online (non-local) authentication was done it would be possible to switch to an only local password authentication. But I think it would be good the hear @ikerexxe and @abbra opinion here as well.

Given the special role of password authentication and if only local methods together with passwords are really needed I think a new option value like e.g. `local_only_with_password` would be better than changing the semantic of `only`.

I agree that for testing passkey authentication it might be more flexible to typically use `local_auth_policy = enable:passkey` and only have a few dedicated tests for the `local_auth_policy = only` code path.

bye,
Sumit

Reply:
> I would say the behavior is correct because password authentication with AD is initially not a local authentication method. Only if it is allowed to store the password hash locally and a successful online (non-local) authentication was done it would be possible to switch to an only local password authentication. But I think it would be good the hear @ikerexxe and @abbra opinion here as well.

I agree with you that if AD password authentication doesn't happen locally, then it makes sense that `local_auth_policy = only` doesn't allow any password authentication.

> I agree that for testing passkey authentication it might be more flexible to typically use `local_auth_policy = enable:passkey` and only have a few dedicated tests for the `local_auth_policy = only` code path.

Makes sense.

Reply:
I agree with @sumit-bose. There will be only one special use case where a Kerberos authentication should be considered local: when the KDC address resolves to the same machine. This is to allow local KDC operation (and something like IPA DC or Samba AD DC operation). In those cases SSSD can reach to the KDC because it is running right on the same system, so it is always available locally.

Reply:
Thank you for all explaining this, test cases are updated and working as expected.
I am marking this as resolved conversation.